packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -479,6 +479,20 @@ class Model(_MyMutableMapping):
             )
         else:
             non_attr_kwargs = [k for k in kwargs if k not in self._attr_to_rest_field]
+            {% if code_model.options["models_mode"] == "dpg" and code_model.options["azure_arm"] %}
+            if non_attr_kwargs:
+                if "properties" in self._attr_to_rest_field:
+                    properties_class = self._attr_to_rest_field["properties"]._class_type
+                    properties_attr_dict = getattr(properties_class, "__dict__", {})
+                    properties_kwargs = {}
+                    for item in non_attr_kwargs:
+                        if item in properties_attr_dict:
+                            properties_kwargs[item] = kwargs[item]
+                    kwargs["properties"] = properties_class(**properties_kwargs)
+                    for item in properties_kwargs:
+                        kwargs.pop(item)
+                        non_attr_kwargs.remove(item)
```

Comment:
My thinking is we don't need to add code to allow users to create a flattened resource model. The service really should create all of these for them, and the main issue really is just access. cc @lmazuel @tadelesh @johanste for your thoughts

Reply:
While it is certainly possible that someone passed in "raw" dicts using their flattened form (and thus it would be/is breaking to remove it), let's start without making the code too complicated.

Reply:
+1 with @johanste @iscai-msft , let's start with an access strategy first

Reply:
It is for compatibility. If we remove the logic here, existing users may be broken in the following scenario:

https://github.com/Azure/autorest.python/blob/fe80efe0ec224b179e8d110a87de6d21d2daf61b/packages/typespec-python/test/azure/mock_api_tests/test_mgmt_models_mode.py#L49-L52

Reply:
As discussed in sync up, if user init with kwargs for flatten properties, then no error will happen, but the real payload will not include it, which cause runtime error from service that is not easy to diagnose. So, +1 with @msyyc 
Another thing I'm concerned is for type checking, will this code fail for mypy because `Catalog` actually does not has `provisioning_state`? This concern applies for all flatten properties' access and write.

Reply:
So 3 scenarios. Assuming `foo` is on `properties`:
``` python
m = VMCreateParameter(foo=42)
m.foo = 42
print(m.foo)
```

One questions: What does the first one does today if I used an attribute that doesn't exist? Silence? Exception?

I have concerns about not trying to get(2) working, as Python would move from doing something, to actually doing nothing at all and let Azure complains. And the error message from the server like "Foo was not set' may be super confusing, especially when the code worked before untouched. @johanste I think we need to think this through a bit more.